resources/prometheus/prometheus-rules.yaml

```diff
@@ -454,6 +454,78 @@ spec:
+          record: central:grpc_server_handling_seconds:rate10m:p90:sli
+        - expr: |
+            sum_over_time(central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90[28d]) / count_over_time(central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90[28d])
+          record: central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:sli
+
+        # - Queries the error rate or the ratio of the instances of central:xxx:rate10m:p90
+        #   were equal to 0 over the total instances of central:xxx:rate10m:p90 within a period.
+        - expr: |
+            1 - central:grpc_server_handling_seconds:rate10m:p90:sli
+          record: central:grpc_server_handling_seconds:rate10m:p90:error_rate28d
+        - expr: |
+            1 - central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:sli
+          record: central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:error_rate28d
+
+        # - Queries error rate for a 1h window.
+        - expr: |
+            1 - (sum_over_time(central:grpc_server_handling_seconds:rate10m:p90[1h]) / count_over_time(central:grpc_server_handling_seconds:rate10m:p90[1h]))
+          record: central:grpc_server_handling_seconds:rate10m:p90:error_rate1h
+        - expr: |
+            1 - (sum_over_time(central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90[1h]) / count_over_time(central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90[1h]))
+          record: central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:error_rate1h
+
+        # - Queries the error budget exhaustion (or consumption) for the whole slo window (28d).
+        - expr: |
+            (1 - central:grpc_server_handling_seconds:rate10m:p90:sli) / 0.01
+          record: central:grpc_server_handling_seconds:rate10m:p90:error_budget_exhaustion28d
+        - expr: |
+            (1 - central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:sli) / 0.01
+          record: central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:error_budget_exhaustion28d
+
+        # - Queries error budget burn rate (a.k.a. burn rate) is the ratio of central:xxx:rate10m:p90:error_rateyyy
+        #   over the error budget for a period (e.g. 1h, 1d, etc).
+        - expr: |
+            central:grpc_server_handling_seconds:rate10m:p90:error_rate1h / 0.01
+          record: central:grpc_server_handling_seconds:rate10m:p90:burn_rate1h
+        - expr: |
+            central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:error_rate1h / 0.01
+          record: central:http_incoming_request_duration_histogram_seconds_bucket:rate10m:p90:burn_rate1h
+
+        # - A sample count filter that ignores central:xxx:rate10m:p90 instances that has samples less than the expected sample count.
```

Comment:
How does the filter treat periods where there is no incoming traffic (and the base metrics are therefore undefined)?